crates/pop-cli/src/common/helpers.rs

```diff
@@ -0,0 +1,302 @@
+// SPDX-License-Identifier: GPL-3.0
+
+use anyhow;
+use std::{
+	fs::{read_to_string, write},
+	path::{Path, PathBuf},
+};
+use toml_edit::{value, DocumentMut, Item, Value};
+
+/// This function is used to determine if a Path is contained inside a workspace, and returns a PathBuf to the workspace Cargo.toml if found
+/// # Parameters
+/// - target_dir: A directory that may be contained inside a workspace
+/// # Output
+/// Option<PahBuf>: Some if the input is contained in a workspace, the inner PathBuf points to the workspace Cargo.toml. Otherwise, None
+pub(crate) fn find_workspace_toml(target_dir: &Path) -> Option<PathBuf> {
+	let mut dir = target_dir;
+	while let Some(parent) = dir.parent() {
+		let cargo_toml = dir.join("Cargo.toml");
```

Comment:
The bug mentioned above is fixed if I change this line: 
```
let cargo_toml = parent.join("Cargo.toml");
```
However, I'm not sure why this issue only affects adding contracts, while adding a pallet works as expected.

Reply:
This is clearly a mistake in my PR 🤪I did wanna use the line you mentioned of course. Fixing it